.github/workflows/check-catalyst.yaml

```
@@ -432,11 +464,16 @@ jobs:
     - name: Checkout Catalyst repo
       uses: actions/checkout@v4
 
+    - name: Set up Python # Ensure python3.10 is used
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ needs.constants.outputs.primary_python_version }}
+
     - name: Install Deps
       run: |
         sudo apt-get update
         sudo apt-get install -y python3 python3-pip libomp-dev libasan6 make ninja-build
```

Do we need these with the setup-python action above?
```suggestion
        sudo apt-get install -y libomp-dev libasan6 make ninja-build
```